parser-utils/src/lib.rs

```diff
@@ -105,3 +105,85 @@ macro_rules! def_value_satisfy_parser_fn {
         });
     }
 }
+
+/// A `ValueParseError` is a `combine::primitives::ParseError`-alike that implements the `Debug`,
+/// `Display`, and `std::error::Error` traits.  In addition, it doesn't capture references, making
+/// it possible to store `ValueParseError` instances in local links with the `error-chain` crate.
+///
+/// This is achieved by wrapping slices of type `&'a [edn::Value]` in an owning type that implements
+/// `Display`; rather than introducing a newtype like `DisplayVec`, we re-use `edn::Value::Vector`.
+#[derive(PartialEq)]
+pub struct ValueParseError {
+    pub position: usize,
+    // Think of this as `Vec<Error<edn::Value, DisplayVec<edn::Value>>>`; see above.
+    pub errors: Vec<combine::primitives::Error<edn::Value, edn::Value>>,
+}
+
+impl std::fmt::Debug for ValueParseError {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f,
+               "ParseError {{ position: {:?}, errors: {:?} }}",
+               self.position,
+               self.errors)
+    }
+}
+
+impl std::fmt::Display for ValueParseError {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        try!(writeln!(f, "Parse error at {}", self.position));
+        combine::primitives::Error::fmt_errors(&self.errors, f)
+    }
+}
+
+impl std::error::Error for ValueParseError {
+    fn description(&self) -> &str {
+        "parse error parsing EDN values"
+    }
+}
+
+impl<'a> From<combine::primitives::ParseError<&'a [edn::Value]>> for ValueParseError {
+    fn from(e: combine::primitives::ParseError<&'a [edn::Value]>) -> ValueParseError {
+        ValueParseError {
+            position: e.position,
+            errors: e.errors.into_iter().map(|e| e.map_range(|r| {
```

Comment:
I confess that I'm a little surprised to have `errors` be something we can move, but I guess that helps!

I was worried that we'd have copying on the error path, where the error might be due to OOM…